marsappspa/src/App.tsx

```diff
@@ -88,20 +93,44 @@ function Component(props: Props) {
 }
 
 function App() {
+  const countString = localStorage.getItem("count");
+  const [count, setCount] = useState(countString == null ? 0 : parseInt(countString));
+
+  useEffect(() => {
+      document.title = `You clicked ${count} times`;
+      localStorage.setItem("count", count.toString());
+  });
   return (
     <div className="App">
       <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <Component name="NASA" p1="fafargigmfaefasrfa" p2="fafaffeafa3f3asfdafadfa" image={nasaLogo} />
-        <Comp1 />
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
+        <Context.Provider value={{count, setCount}}>
+          <Router>
+            <div>
+              <nav>
+                <ul>
+                  <li>
+                    <Link to="/">Home</Link>
+                  </li>
+                  <li>
+                    <Link to="/NASA">NASA</Link>
+                  </li>
+                  <li>
+                    <Link to="/Button">Button</Link>
+                  </li>
+                  <li>
+                    <Link to="/ComplexButton">Complex Button</Link>
+                  </li>
+                </ul>
+              </nav>
+              <Routes>
+                <Route path="/" element={<ReactBase />} />
+                <Route path="/NASA" element={<Component name="NASA" p1="fafargigmfaefasrfa" p2="fafaffeafa3f3asfdafadfa" image={nasaLogo} />} />
```

Comment:
let's use React composition instead of passing the text for the paragraphs

https://felixgerschau.com/react-component-composition/